### ISSUE TITLE
Add Location Finder to ENgage

### ISSUE THREAD

**Issue by ncadjan** [arckinteractive/en-issues#34]
Our CEO wants a function to locate members in a specific city.  Can you tell us if it is possible and what it would take to locate all members in a specific city?  We have data in SF and in ENsight.  However, that data may be their corporate office instead of their physical location.  If we had a field we entered in SF or ENsight, could we start typing a city location, have the members select it and then see a list of all members in that location? 

Or, can we do a find member from a current location?  I don't think this is wise or that members will enable location services so this is probably a less-desirable option but since our CEO will ask for it, I want to know what it would take and if it is suggested. Also, if a person is looking to see who is in the city they are going to, it doesn't make sense to me because it would be based on the location they are going to, not the one they are at. 

My husband's company has an app for locating graves that I think might be an idea to start with.  https://www.findagrave.com/mobileapp/  Just an FYI because you may know of an app plugin that already does a location-based search.

**Comment by paulstewart** [comment]
We can do both. In the case that a member tries to do a search from their location and we can't determine their location, the app would simply prompt them to enter a physical address.

Would you like to see a proposal/quote for this?

**Comment by ncadjan** [comment]
Yes please.  This isn’t a hurry project.  We are looking ahead and answering the question of feasibility and cost for our CEO.

If there are optional ways to do this, could you list them separately and the cost for a basic and a more advanced feature set?

**Comment by paulstewart** [comment]
For reference:

- https://www.raymondcamden.com/2015/05/18/tracking-and-notifying-geolocation-status-with-ionic
- https://www.thepolyglotdeveloper.com/2015/09/support-ibeacons-in-your-ionic-framework-mobile-app/
- https://developers.google.com/maps/articles/phpsqlsearch_v3#findnearsql

**Comment by ncadjan** [comment]
Thanks.  This seems like the members would all have to allow Locations on the app right?

**Comment by beck24** [comment]
Yes they would have to have location enabled.  We could detect if it's not enabled and prompt them to enable it.

**Comment by ncadjan** [comment]
But could we pull a location field from ENsight and populate lists like that?

**Comment by beck24** [comment]
We could do one or the other, trying to juggle both would probably be complicated

**Comment by ncadjan** [comment]
Thanks.  Is it easy to bid each separately?  I know the CEO will ask if we can do the nifty thing by location but I doubt our members will allow locations to be turned on.  These are private people.  I wouldn’t and I don’t in any apps other than those specific ones I need it for (Maps, etc).

**Comment by beck24** [comment]
We could add a new location profile field (that isn't necessarily visible on the profile to non-admin) that would be geo-searchable.  Then in the app we could reference that field for proximity searches.

**Comment by ncadjan** [comment]
That is probably the least expensive right?

**Comment by beck24** [comment]
It's probably about the same either way.  Ultimately we have to store that location data on ENsight, then the search is the same.  All that's changing is how we get that info - via app detection or manual entry

**Comment by ncadjan** [comment]
Thanks.

**Comment by ncadjan** [comment]
Matt--I don't see a bid here.  This might raise its head again.  Would you please bid?  I know we had a phone conversation about this too.

**Comment by beck24** [comment]
I'm going to say 12hrs for the profile field input with proximity search.
Some questions will need to be answered though:

Would they need to turn on location to do the proximity search, or would you want them to put in an address?  With location services, they can then see who is close to them now.  With manual input though, it opens things up to possible abuse as people can just run proximity searches from anywhere.  Depends on what expectations are.

If we want to track location from the app, and keep it updated it brings up how often we refresh that location.  Should that be a background thing, or a manually triggered event?
